frontend/lib/editor/common.ts

```diff
@@ -0,0 +1,61 @@
+import { type WritableComputedRef } from 'vue'
+
+export interface ForUse<T> {
+  isRequired: boolean
+  isCompleted: ComputedRef<boolean>
+  value: WritableComputedRef<T>
+}
+
+export enum Validation {
+  NotEmpty = 'NotEmpty',
+  AlphanumericAndDashesAndUnderscores = 'AlphanumericAndDashesAndUnderscores',
+}
```

Comment:
This all seems like a well-structured library, though I worry we're reinventing the wheel along a few different axes here:

- These kind of form validations are built into HTML5, see https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation#using_built-in_form_validation
- AFAICT, our requirements here are not unique and we could probably benefit from an off-the-shelf solution here.
	- For example, [PrimeVue has integration with VeeValidate](https://primevue.org/inputtext/#form), a popular Vue validation library (which should work fine with Nuxt)

Reply:
This is a fair criticism. Since this has changed quite a bit, and I think simplified quite a bit, would you take another look before I go too far into the Vee-Validate rabbit hole? 

Context: I've messed around with validation libraries in the past, and my prior experiences with them (most recently in abound) have always been poor - they do that thing we always talk about of offering leaky abstractions that aren't actually solving for the complexity they try to tackle.